File: bits_simulators_portal_frontend/src/app/pages/atm/model/OperationAtmDto.ts
import {ScriptCasesDefinitionDto} from "./ScriptCasesDefinition";
import {AtmCardProfilDto} from "./AtmCardProfilDto";

export interface OperationAtmDto {
  id: number;
  libelle: string;
  scriptCasesDefinitions: ScriptCasesDefinitionDto[];
  atmCardProfilDto: AtmCardProfilDto;
  bankCode: string;
}

File: bits_simulators_portal_backend/src/main/java/com/simulator/atm/dto/OperationAtmDto.java
package com.simulator.atm.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OperationAtmDto implements Serializable {
    Long id;
    String libelle;
    List<ScriptCasesDefinitionDto> scriptCasesDefinitions;
    AtmCardProfilDto atmCardProfilDto;
    String bankCode;
}
